src/renderer/components/common/QrCode/QrReader/QrReader.tsx

```diff
@@ -184,9 +190,14 @@ const QrReader = ({
         continue;
       }
 
-      const result = EXPORT_ADDRESS.decode(fountainResult.slice(3));
+      if (isDynamicDerivations) {
+        const result = DYNAMIC_DERIVATIONS_ADDRESS_RESPONSE.decode(fountainResult.slice(3));
+        onResult?.(makeResultPayload(result));
+      } else {
+        const result = EXPORT_ADDRESS.decode(fountainResult.slice(3));
+        onResult?.(makeResultPayload(result.payload));
```

Comment:
`const result = EXPORT_ADDRESS.decode(fountainResult.slice(3)).payload`

and move `onResult?.(makeResultPayload(result));` outside of if-else